### PR TITLE
Start API server only once

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

**Comment by dgageot** [comment]
WDYT? @nkubala

File: cmd/skaffold/app/cmd/cmd.go [hunk 1]
```diff
@@ -44,47 +46,68 @@ var (
...
-			go func() {
-				if err := updateCheck(updateMsg); err != nil {
-					logrus.Infof("update check failed: %s", err)
-				}
-			}()
-		}
+			// Start API Server
+			if cmd.Use == "dev" {
+				// TODO(dgageot): api server is always started in dev mode, right now.
+				// It should instead default to true.
```
Comment (priyawadhwa) [inline cmd/skaffold/app/cmd/cmd.go]: do you mean default to false, if the api server is always started right now?

Comment (dgageot) [reply cmd/skaffold/app/cmd/cmd.go]: I mean it should default to true for dev mode but the user should be able to disable it. Currently, it's always on for dev mode

Comment (priyawadhwa) [reply cmd/skaffold/app/cmd/cmd.go]: I see, thanks!

Comment (nkubala) [reply cmd/skaffold/app/cmd/cmd.go]: how about either
* changing the `--enable-rpc` flag to a `*bool`, so we can tell if the user didn't provide the flag (vs. explicitly providing `--enable-rpc=false`, or
* adding a separate flag `--disable-rpc`, and checking that here e.g. `if cmd.Use == "dev" && opts.DisableRPC == false`. we'd have to have a way to make sure users don't set `--enable-rpc` and `--disable-rpc` at the same time though (don't know if there's an easy way to do this using cobra)

I'm fine with this being out of scope for this PR though.

**Pull request merged**